provisioning/image_cache.go

```diff
@@ -129,6 +147,35 @@ func createContainerImageCache(images *Images) corev1.Container {
 	return container
 }
 
+func newImageCacheInitContainers(info *ProvisioningInfo) ([]corev1.Container, error) {
+	initContainers := []corev1.Container{}
+	// If the PreProvisioningOSDownloadURLs are set, we fetch the URLs of either CoreOS ISO and IPA URLs or in some
+	// cases only the IPA URLs
+	liveURLs := getPreProvisioningOSDownloadURLs(&info.ProvConfig.Spec)
+	if liveURLs != nil {
+		newURLs, err := getTransformedURLs(liveURLs, info.Namespace)
+		if err != nil {
+			return nil, err
+		}
+		initContainers = append(initContainers, createInitContainerMachineOsDownloader(info, strings.Join(newURLs, ","), true, false))
+	}
+	// Download the transformed URL containing qcow2
+	if info.ProvConfig.Spec.ProvisioningOSDownloadURL != "" {
+		origURLs := []string{info.ProvConfig.Spec.ProvisioningOSDownloadURL}
+		newURLs, err := getTransformedURLs(origURLs, info.Namespace)
```

Comment:
nit: could just call transformURL here.

Reply:
+1. I tried to build a helper but it was not so useful. Removing it. Thanks.